feat(cli): add --var shorthand for --variable

ammario opened the pull request:
feat(cli): add --var shorthand for --variable

`--variable` is used frequently enough to deserve a shorthand. Unfortunately,
`-v` is taken by verbose, and `-V` is too easily confused with version or
verbose, so we're left with "--var".

ericpaulsen reviewed (approved):
docs & helm files look good.

ammario commented:
thx @ericpaulsen

I'll hold off on merging this until @mtojek gives the 👍🏽 on the flag name.

mtojek commented on cli/templatecreate.go:
```diff
@@ -166,9 +166,10 @@ func (r *RootCmd) templateCreate() *clibase.Cmd {
+			Flag:          "variable",
+			Description:   "Specify a set of values for Terraform-managed variables.",
+			Value:         clibase.StringArrayOf(&variables),
+			FlagShorthand: "V",
```
`-V` is always confusing as it usually means `--version` or `--verbose`.

Maybe we should another alias, `-var`, instead? Alternatively, we can add support for [TF_VAR_](https://github.com/coder/coder/issues/8279).

mtojek reviewed (approved):
I'm not blocking this PR, but I admit that `-V` may be confusing.

ammario committed:
feat(cli): add --var shorthand for --variable

`--variable` is used frequently enough to deserve a shorthand. Unfortunately,
`-v` is taken by verbose, and `-V` is too easily confused with version or
verbose, so we're left with "--var".
